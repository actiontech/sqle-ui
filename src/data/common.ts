export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}