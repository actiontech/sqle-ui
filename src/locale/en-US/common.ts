// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: 'username',
  password: 'password',

  unknownError: 'Unknown Error',

  ok: 'Ok',
  cancel: 'Cancel',
  submit: 'Submit',
  close: 'Close',
  edit: 'Edit',
  delete: 'Delete',
  reset: 'Reset',
  search: 'Search',
  retry: 'Retry',
  back: 'Back',
  upload: 'Upload',
  resetAndClose: 'Reset Form And Close',
  operateSuccess: 'Operate successfully',
  prevStep: 'Prev',
  nextStep: 'Next',

  expansion: 'Expand',
  collapse: 'Collapse',

  theme: {
    light: 'Light Theme',
    dark: 'Dark Theme',
  },

  logout: 'Logout',
  account: 'Account',

  nav: {
    title: 'SQLe',
  },

  request: {
    noticeFailTitle: 'Request Failed',
  },

  form: {
    placeholder: {
      input: 'Please input {{name}}',
      select: 'Please select {{name}}',
      searchInput: 'Input the {{name}} to search',
      searchSelect: 'Select the {{name}} to search',
    },
    rule: {
      require: 'Please input {{name}}',
      passwordNotMatch: 'The two passwords that you entered do not match!',
      email: 'The input is not valid E-mail!',
    },
  },
};
