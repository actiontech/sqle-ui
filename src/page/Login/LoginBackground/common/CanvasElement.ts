import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
