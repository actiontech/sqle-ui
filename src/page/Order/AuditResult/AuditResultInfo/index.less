@import '../../../../styles/_variable.less';

.result-box {
  border-radius: 4px;
  padding: 2px 5px;
  display: flex;
  align-items: center;
  .result-box-level {
    margin-left: 5px;
    width: 110px;
    display: inline-box;
  }
  &-passed {
    color: @text-green;
    border: 1px solid @text-green;
  }
  &-warn {
    color: @text-orange;
    border: 1px solid @text-orange;
  }
  &-error {
    color: @text-red;
    border: 1px solid @text-red;
  }
  &-notice {
    color: @text-blue;
    border: 1px solid @text-blue;
  }
}

.result-box-list {
  display: inline-flex;
  gap: 8px;
  flex-direction: column;
}
