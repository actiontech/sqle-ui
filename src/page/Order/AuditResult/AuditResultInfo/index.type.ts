import {AuditResultErrorMessageProps} from '../../../../components/AuditResultErrorMessage/index.type'

export interface AuditResultColumnProps extends AuditResultErrorMessageProps {}