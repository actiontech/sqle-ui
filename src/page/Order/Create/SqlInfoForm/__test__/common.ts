import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
