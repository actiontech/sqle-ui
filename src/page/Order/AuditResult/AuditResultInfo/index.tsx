import { useMemo } from 'react';
import { AuditResultColumnProps } from './index.type';
import AuditResultErrorMessage from '../../../../components/AuditResultErrorMessage';
import {
  InfoCircleOutlined,
  CheckOutlined,
  WarningOutlined,
} from '@ant-design/icons';
import { RuleResV1LevelEnum } from '../../../../api/common.enum';
import './index.less';

const AuditResultInfo: React.FC<AuditResultColumnProps> = (props) => {
  const auditResult = useMemo(
    () => props?.auditResult ?? [],
    [props.auditResult]
  );
  const auditResultNum = auditResult.length;

  const getClassNameAndIconByStatus = (status: string) => {
    const defaultClassName = 'result-box';
    const defaultIcon = <InfoCircleOutlined />;

    switch (status) {
      case 'passed':
        return {
          className: `${defaultClassName}-success`,
          icon: <CheckOutlined />,
        };
      case RuleResV1LevelEnum.normal:
      case RuleResV1LevelEnum.warn:
        return {
          className: `${defaultClassName}-warning`,
          icon: <WarningOutlined />,
        };
      case RuleResV1LevelEnum.error:
        return {
          className: `${defaultClassName}-error`,
          icon: <InfoCircleOutlined />,
        };
      default:
        return {
          className: '',
          icon: defaultIcon,
        };
    }
  };

  const renderResultBox = (resultLevel: string, resultNum?: number) => {
    const { className, icon } = getClassNameAndIconByStatus(resultLevel);

    return (
      <div className={`result-box ${className}`}>
        {icon}
        <span className="result-box-level">{resultLevel}</span>
        {resultNum && <span>{resultNum}</span>}
      </div>
    );
  };

  const getResultNum = (status: string) =>
    auditResult.filter((item) => item.level === status).length;

  const renderResultBoxList = () => {
    const renderLevelKey = [
      RuleResV1LevelEnum.error,
      RuleResV1LevelEnum.warn,
      RuleResV1LevelEnum.normal,
      RuleResV1LevelEnum.notice,
    ];

    return (
      <div className="result-box-list">
        {renderLevelKey.map((key) => {
          const innerResultNum = getResultNum(key);
          return (
            !!innerResultNum && (
              <li key={key}>{renderResultBox(key, innerResultNum)}</li>
            )
          );
        })}
      </div>
    );
  };

  const renderAuditColumn = () => {
    if (auditResultNum === 1)
      return <AuditResultErrorMessage auditResult={auditResult} />;

    if (auditResultNum === 0) return renderResultBox('passed');

    return renderResultBoxList();
  };

  return <>{renderAuditColumn()}</>;
};

export default AuditResultInfo;
