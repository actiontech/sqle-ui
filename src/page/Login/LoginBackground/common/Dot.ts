import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
