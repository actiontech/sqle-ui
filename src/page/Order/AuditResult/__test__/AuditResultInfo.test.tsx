import { render, screen } from '@testing-library/react';
import {
  getAllBySelector,
  getBySelector,
} from '../../../../testUtils/customQuery';
import AuditResultInfo from '../AuditResultInfo';
import { IAuditResult } from '../../../../api/common';
import { mockUseStyle } from '../../../../testUtils/mockStyle';

describe('Order/AuditResult/AuditResultInfo', () => {
  beforeEach(() => {
    mockUseStyle();
  });
  const multipleAuditResults: IAuditResult[] = [
    { level: 'error', message: 'schema 不存在', rule_name: '' },
    {
      level: 'error',
      message:
        '禁止使用没有where条件的sql语句或者使用where 1=1等变相没有条件的sql',
      rule_name: 'all_check_where_is_invalid',
    },
    {
      level: 'warn',
      message: 'select 语句必须带limit,且限制数不得超过1000',
      rule_name: 'dml_check_select_limit',
    },
    {
      level: 'normal',
      message: 'Test normal msg',
      rule_name: 'dml_check_select_limit',
    },
    {
      level: 'notice',
      message: 'Test notice msg',
      rule_name: '',
    },
  ];

  const singleAuditResult: IAuditResult[] = [
    { level: 'error', message: 'schema 不存在', rule_name: '' },
  ];

  test('should render audit result overview when get more than 2 audit results', () => {
    const { container } = render(
      <AuditResultInfo auditResult={multipleAuditResults} />
    );
    expect(container).toMatchSnapshot();
    expect(getAllBySelector('.result-box')).toHaveLength(4);

    expect(screen.getByText('error')).toBeInTheDocument();
    expect(getBySelector('.result-box-error :nth-child(3)').innerHTML).toBe(
      '2'
    );

    expect(screen.getByText('warn')).toBeInTheDocument();
    expect(
      getBySelector(':nth-child(3)', getAllBySelector('.result-box-warn')[0])
        .innerHTML
    ).toBe('1');

    expect(screen.getByText('normal')).toBeInTheDocument();

    expect(screen.getByText('notice')).toBeInTheDocument();
    expect(
      getBySelector(':nth-child(3)', getAllBySelector('.result-box-notice')[0])
        .innerHTML
    ).toBe('1');
  });

  test('should render single audit result detail when only get one audit result', () => {
    const { container } = render(
      <AuditResultInfo auditResult={singleAuditResult} />
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText('schema 不存在')).toBeInTheDocument();
  });

  test('should render passed when there is no audit result', () => {
    const { container } = render(<AuditResultInfo auditResult={[]} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('passed')).toBeInTheDocument();
  });
});
