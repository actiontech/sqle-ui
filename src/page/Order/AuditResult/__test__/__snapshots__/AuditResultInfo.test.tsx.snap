// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order/AuditResult/AuditResultInfo should render audit result overview when get more than 2 audit results 1`] = `
<div>
  <div
    class="result-box-list"
  >
    <li>
      <div
        class="result-box result-box-error"
      >
        <span
          aria-label="info-circle"
          class="anticon anticon-info-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="info-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
        <span
          class="result-box-level"
        >
          error
        </span>
        <span>
          2
        </span>
      </div>
    </li>
    <li>
      <div
        class="result-box result-box-warning"
      >
        <span
          aria-label="warning"
          class="anticon anticon-warning"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="warning"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M464 720a48 48 0 1096 0 48 48 0 10-96 0zm16-304v184c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V416c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zm475.7 440l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zm-783.5-27.9L512 239.9l339.8 588.2H172.2z"
            />
          </svg>
        </span>
        <span
          class="result-box-level"
        >
          warn
        </span>
        <span>
          1
        </span>
      </div>
    </li>
    <li>
      <div
        class="result-box result-box-warning"
      >
        <span
          aria-label="warning"
          class="anticon anticon-warning"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="warning"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M464 720a48 48 0 1096 0 48 48 0 10-96 0zm16-304v184c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V416c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zm475.7 440l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zm-783.5-27.9L512 239.9l339.8 588.2H172.2z"
            />
          </svg>
        </span>
        <span
          class="result-box-level"
        >
          normal
        </span>
        <span>
          1
        </span>
      </div>
    </li>
    <li>
      <div
        class="result-box "
      >
        <span
          aria-label="info-circle"
          class="anticon anticon-info-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="info-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
        <span
          class="result-box-level"
        >
          notice
        </span>
        <span>
          1
        </span>
      </div>
    </li>
  </div>
</div>
`;

exports[`Order/AuditResult/AuditResultInfo should render passed when there is no audit result 1`] = `
<div>
  <div
    class="result-box result-box-success"
  >
    <span
      aria-label="check"
      class="anticon anticon-check"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="check"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
        />
      </svg>
    </span>
    <span
      class="result-box-level"
    >
      passed
    </span>
  </div>
</div>
`;

exports[`Order/AuditResult/AuditResultInfo should render single audit result detail when only get one audit result 1`] = `
<div>
  <div
    class="ant-space ant-space-vertical audit-result-error-message-wrapper"
  >
    <div
      class="ant-space-item"
    >
      <div
        class="ant-row ant-row-no-wrap"
      >
        <div
          class="ant-col"
          style="flex: 0 0 50px;"
        >
          <div
            class="ant-space ant-space-vertical ant-space-align-center sqle-rule-icon"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 1px;"
            >
              <div>
                <span
                  aria-label="info-circle"
                  class="anticon anticon-info-circle"
                  role="img"
                  style="font-size: 25px; color: rgb(240, 0, 0);"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="info-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography ant-typography-secondary"
              >
                error
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col message"
          style="flex: 1 1 auto;"
        >
          <span>
            schema 不存在
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
